[quant][graphmode] refactor nodeQuantizable

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #33481 [quant][graphmode] Handling ops doesn't require observation in insertObservers
* #33532 [quant][graphmode] Add quantization pattern for quantized::add_relu
* #33279 [quant][graphmode] Add quantized conv2d-relu fusion pattern
* #32816 [quant][graphmode] Add add_relu pattern in skip values
* #32814 [quant][graphmode] Skip quantizing input and output in matched module
* #33277 [WIP][quant][graphmode] Observing input/output values in callsite
* #33649 [quant][graphmode][refactor] Factor out getInvokedMethod
* #33276 [quant][graphmode][refactor] Move check for weight outside of insertObserverFor
* #33275 [quant][graphmode][refactor] Move values_to_skip check inside valueNeedsToBeQuantized
* #33274 [quant][graphmode][refactor] Simplify signature for insertObserverFor
* #33273 [quant][graphmode][refactor] Checks for bias and weight
* #33173 [quant][graphmode] Add a pass to handle ops that doesn't require observation
* #33531 [quant][graphmode] Replicate dequantize nodes
* #33172 [quant][graphmode][refactor] Factor out insertDequantCall
* **#33171 [quant][graphmode] refactor nodeQuantizable**

Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20087845](https://our.internmc.facebook.com/intern/diff/D20087845)

jerryzh168 committed:
[quant][graphmode] refactor nodeQuantizable

Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

dreiss commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -106,22 +106,24 @@ std::string getFuncName(Value* func_value) {
+                    const std::vector<std::string>& call_funcs,
+                    const std::vector<std::string>& aten_funcs) {
+  std::vector<Symbol> all_funcs;
   std::transform(
```
Could this be written as

```
for (const auto& funcs : {call_funcs, aten_funcs}) {
  for (const auto& s : funcs) {
    all_funcs.push_back(s);
  }
}
```

dreiss replied:
I mean `push_back(Symbol::aten(s))`

jerryzh168 replied:
changed the logic a little bit, I'm going to list out both call functions and aten functions separately.

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] refactor nodeQuantizable"


Summary:
For better code reuse

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20087845](https://our.internmc.facebook.com/intern/diff/D20087845)

[ghstack-poisoned]
